src/functional/fencil_processors/source_modules/cpp_gen.py

```diff
@@ -14,15 +14,35 @@
 
 
 import ctypes
+from dataclasses import dataclass
 from types import MappingProxyType
 from typing import Final, Sequence, Type
 
 import numpy
 
-from functional.fencil_processors import source_modules
+from eve.codegen import format_source
+from functional.fencil_processors.source_modules import source_modules
 
 
-LANGUAGE_ID: Final = "cpp"
+@dataclass(frozen=True)
+class CppLanguage:
+    """Implements source_modules.IncludeImplementationLanguageProtocol for C++ language settings."""
+
+    name: str
+    implementation_extension: str
+    include_extension: str
+    formatting_style: str
```

Comment:
Maybe a comment on where to find the available options for formatting styles? Perhaps an enum in eve?

Reply:
I will leave this up to eve...

Reply:
I put that into the docstring of `source_modules.LanguageSettings`